### PR TITLE
fix: #221 filtered issues in archived repositories

### PR DESCRIPTION
**Pull request by yokishava**
# Summary
I've tried to filter archived pull requests before, but I saw this issue
We have done the same for issues in archived repositories.

#221

I also saw a comment that the code was becoming redundant. I considered modifying the code to take that into account, but decided against it because the scope of the change would be too large.

## How did you test this change?
- remove `excludeArvhicedPullRequests` in prsection.go
- remove `sort` in prsection.go and issuesection.go
- filtered issues and sorted issues in FetchIssues
- filtered pull requests and sorted pull requests in FetchPullRequests
- remove test.yaml in github workflows

## Images/Videos

before
スクリーンショット 2023-01-17 19 14 08

after
スクリーンショット 2023-01-17 19 14 44

I used this repository to check if archived issues are filtered.
https://github.com/yokishava/element

### CODE REVIEW

**Comment by robdimsdale** [comment]
This is almost exactly what I was planning on doing to address #221 - thank you for beating me to it, @yokishava !

I am curious to learn more about why you removed the test. I was planning on adding a test to `issuesection` (or `issueapi`) similar to what you previously added for PRs. But I see you removed it.

I think there's still value in testing the archived/sort functionality - do you agree ? If so, should we add the test back in?

**Comment by yokishava** [comment]
Thank you for comment!

The reason for removing the tests is that the changes made it harder to write tests.

You are right that it would be better to have tests, but if we are going to write tests, I think we need to make more changes to FetchIssue and FetchPullRequests to get there. I have lowered the priority of the tests to reduce the scope of the changes.

If you still think it would be better to have tests, I would be happy to modify them too.

**Comment by robdimsdale** [comment]
> The reason for removing the tests is that the changes made it harder to write tests.

Ah, yes. The `issueapi`/`prapi` is tightly coupled to the GitHub api, so in order to test the archiving and sort functionality, we'd either have to refactor the relevant functions to consume a mockable interface, or we'd have to commit to integration-testing against a real, archived, repository.

I agree that both of those seem out of scope for this change, so I agree with removing the tests.

**Comment by dlvhdr** [comment]
I'm good with this as well.

File: data/issueapi.go [hunk 1]
```diff
@@ -101,8 +99,16 @@ func FetchIssues(query string, limit int, pageInfo *PageInfo) (IssuesResponse, e
 
 	issues := make([]IssueData, 0, len(queryResult.Search.Nodes))
 	for _, node := range queryResult.Search.Nodes {
+		if node.Issue.Repository.IsArchived {
+			continue
+		}
 		issues = append(issues, node.Issue)
 	}
+
+	sort.Slice(issues, func(i, j int) bool {
```
Comment (dlvhdr) [inline data/issueapi.go:108]: Can you please create an issue that removes the sorting code, and adds a default `sort:updated` like in https://github.com/dlvhdr/gh-dash/pull/223

**Review by dlvhdr — APPROVED** [review]

File: data/issueapi.go (hunk 1 above)
Comment (robdimsdale) [reply data/issueapi.go:108]: I'm not sure I follow the reference - this issue is `#223` and when I searched existing issues and PRs for `sort` I don't see any other issue that I could reference.

Comment (dlvhdr) [reply data/issueapi.go:108]: I swear I pasted a link to the Github filters docs :D

Comment (robdimsdale) [reply data/issueapi.go:108]: done: https://github.com/dlvhdr/gh-dash/issues/225

Comment (yokishava) [reply data/issueapi.go]: Thanks for review!
I added sort qualifier to `makeIssuesQuery` and `makePullRequestsQuery` .

**Pull request merged**